src/Microsoft.Health.Fhir.Shared.Core/Features/Search/BundleFactory.cs

```diff
@@ -93,43 +88,23 @@ public ResourceElement CreateHistoryBundle(SearchResult result)
 
             if (result != null)
             {
-                IEnumerable<Bundle.EntryComponent> entries = result.Results.Select(r =>
-                {
-                    var resource = _deserializer.Deserialize(r);
-                    var hasVerb = Enum.TryParse(r.Request?.Method, true, out Bundle.HTTPVerb httpVerb);
-
-                    return new Bundle.EntryComponent
-                    {
-                        FullUrlElement = new FhirUri(_urlResolver.ResolveResourceUrl(resource, true)),
-                        Resource = resource.Instance.ToPoco<Resource>(),
-                        Request = new Bundle.RequestComponent
-                        {
-                            Method = hasVerb ? (Bundle.HTTPVerb?)httpVerb : null,
-                            Url = hasVerb ? $"{resource.InstanceType}/{(httpVerb == Bundle.HTTPVerb.POST ? null : resource.Id)}" : null,
-                        },
-                        Response = new Bundle.ResponseComponent
-                        {
-                            LastModified = r.LastModified,
-                            Etag = WeakETag.FromVersionId(r.Version).ToString(),
-                        },
-                    };
-                });
+                IEnumerable<Bundle.EntryComponent> entries = result.Results.Select(selectionFunction);
```

Comment:
Any reason to not just pass in IEnumerable<Bundle.EntryComponent> instead of the delegate?

Reply:
I don't see how you could pass in the entries object to the Select method. Select returns a filled list, it doesn't have an overload that takes in a list and fills it.